src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -37,7 +40,20 @@ def _add_message_to_coda(coda, coda_dataset_config, engagement_db_message):
 
     # If the engagement database message already has labels, initialise with these in Coda.
     if len(engagement_db_message.labels) > 0:
-        # TODO: Validate that these labels are valid under the code schemes being copied to.
+        # Ensure the existing labels are valid under the code schemes being copied to, by checking the label's scheme id
+        # exists in this dataset's code schemes or the ws correct dataset scheme, and that the code id is in the
+        # code scheme.
+        valid_code_schemes = [c.code_scheme for c in coda_dataset_config.code_scheme_configurations]
+        valid_code_schemes.append(ws_correct_dataset_code_scheme)
+        valid_code_schemes_lut = {code_scheme.scheme_id: code_scheme for code_scheme in valid_code_schemes}
+        for label in engagement_db_message.labels:
+            assert label.scheme_id in valid_code_schemes_lut.keys(), \
+                f"Scheme id {label.scheme_id} not valid for Coda dataset {coda_dataset_config.coda_dataset_id}"
+            valid_codes = valid_code_schemes_lut[label.scheme_id].codes
+            valid_code_ids = [code.code_id for code in valid_codes]
+            assert label.code_id == "SPECIAL-MANUALLY_UNCODED" or label.code_id in valid_code_ids, \
+                f"Code ID {label.code_id} not found in Scheme {label.scheme_id}"
```

Comment:
```suggestion
                f"Code ID {label.code_id} not found in Scheme {label.scheme_name}" 
```
Much easier to look up while debugging

Reply:
thanks, updated